suite-native/discovery/src/discoveryThunks.ts

```diff
@@ -1,24 +1,247 @@
+        }
+
+        // eslint-disable-next-line no-restricted-syntax
+        for (const bundleItem of descriptorsBundle) {
+            // eslint-disable-next-line no-await-in-loop
+            const { success, payload: accountInfo } = await TrezorConnect.getAccountInfo({
+                coin: bundleItem.coin,
+                descriptor: bundleItem.descriptor,
+            });
+
+            if (success) {
+                if (accountInfo.empty) {
+                    isFinalRound = true;
+                    break;
+                }
+
+                dispatch(
+                    accountsActions.createAccount(
+                        deviceState,
+                        bundleItem,
+                        accountInfo,
+                        false,
+                        // TODO: name of newly created account will be handled in follow-up issue:
+                        // see more: https://github.com/trezor/trezor-suite/issues/9501
+                        bundleItem.descriptor,
+                    ),
+                );
+            }
+        }
+
+        return isFinalRound;
+    },
+);
+
+const discoverNetworkBatchThunk = createThunk(
+    `${DISCOVERY_MODULE_PREFIX}/discoverXpubBatchThunk`,
+    async (
+        {
+            deviceState,
+            device,
+            round = 1,
+            network,
+            mutex,
+        }: {
+            deviceState: string;
+            device: TrezorDevice;
+            round?: number;
+            network: Network;
+            mutex: DeviceAccessMutex;
+        },
+        { dispatch, getState },
+    ) => {
+        const discovery = selectDiscoveryForDevice(getState());
+
+        if (!discovery || !deviceState) {
+            return;
+        }
+
+        const accountType = network.accountType || 'normal';
+
+        const lastDiscoveredAccountIndex = (round - 1) * DISCOVERY_ROUND_SIZE;
+
+        // Skip Cardano legacy/ledger account types if device does not support it.
+        const isIncompatibleCardanoType =
+            network.networkType === 'cardano' &&
+            (network.accountType === 'ledger' || network.accountType === 'legacy') &&
+            !discovery.availableCardanoDerivations?.includes(network.accountType);
+
+        const wasLimitReached = lastDiscoveredAccountIndex >= MAX_ACCOUNTS_PER_NETWORK;
+
+        if (isIncompatibleCardanoType || wasLimitReached) {
+            dispatch(finishNetworkTypeDiscoveryThunk());
+            return;
+        }
+
+        const chunkBundle: Array<DiscoveryItem & { showOnTrezor: boolean }> = [];
+
+        Array.from(Array(DISCOVERY_ROUND_SIZE).keys()).forEach(index => {
+            const accountPath = network.bip43Path.replace(
+                'i',
+                (lastDiscoveredAccountIndex + index).toString(),
+            );
+
+            // Check if account does not already exists.
+            const account = selectAccountByNetworkAndPath(getState(), network.symbol, accountPath);
```

Comment:
account => currentAccount ? 

for the clarity what it represents?

Reply:
change the selector `selectIsAccountAlreadyDiscovered` to return boolean right away. It makes the code clearer and enforces better naming. Fixed in b739e356c3.